test/elixir/test/delayed_commits_test.exs

```diff
@@ -0,0 +1,31 @@
+defmodule DelayedCommitsTest do
+  use CouchTestCase
+
+  @moduledoc """
+  Test CouchDB delayed commits
+  This is a port of the delayed_commits.js suite
+
+  Note that delayed_commits is deprecated in 2.0, so this is a minimal
+  test to show it still works. delayed_commits will be removed in 3.0.
+  """
+
+  @tag config: [
+         {"couchdb", "delayed_commits", "true"}
+       ]
+  @tag :with_db
+  test "delayed commit", context do
+    db_name = context[:db_name]
+    doc_id = "doc-1"
+    resp = Couch.put("/#{db_name}/#{doc_id}", body: %{a: 2, b: 4})
+    assert resp.status_code >= 201
+    assert resp.body["ok"]
+
+    resp = Couch.get("/#{db_name}/#{doc_id}")
+    assert resp.status_code == 200, "The new doc should be in the database"
```

Comment:
thank you for using explanation argument for assertion.

Reply:
Could it be useful to define a bunch of helper functons(`json_response/2`, `text_response/2`) like [Phoenix](https://github.com/phoenixframework/phoenix/blob/4da40a29f968b6e834ab55849af4a7337815ea50/lib/phoenix/test/conn_test.ex#L417) did?
But pass `%HTTPotion.Response{}` instead of `Plug.Conn`.
So we can do something like:

```elixir
resp = Couch.get("/#{db_name}/#{doc_id}")
assert %{"ok" => true} = json_response(resp, 200)
```
or
```elixir
resp = Couch.get("/#{db_name}/#{doc_id}")
assert %{"error" => _error} = json_response(resp, 404)
```

Reply:
In my opinion that would do more harm than good. I believe that writing tests is inherently different from writing "normal" code and in former the simplicity and readability should be always prioritized over tersity.

When I write
```
assert resp.status_code == 200
assert resp.body == ${"ok" => true}
```
I can scan these lines and instantly understand what's going on here.

If i hide it into wrapper
```
assert %{"ok" => true} = json_response(resp, 200)
```
sure I saved one line, but now I have to stop and parse what exactly happening here. Unless I remember exactly what `json_response/2` does I have to guess: is that `200` a status code? Is the function going to raise an exception if status code not `200` or return `false`? Is any body transformation happening "under the hood"?

In general I'd prefer us to introduce code obfuscating changes to tests only when it's absolutely unavoidable.

Reply:
1) I don't find the second example more readable. 2) A thing that'd require a new or occasional test writer to stop and look into documentation or helper's code is an unnecessary hurdle. 3) I don't see a point to discuss here our personal opinion on the test writing. I suggest you to open a separate GH issue on this topic for a broader audience.